### PR TITLE
Add table scaleway_billing_consumption

### PR DESCRIPTION
**Pull request by jplanckeel**
# Context

Add table billing consumption to explore cost usage. 

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  category_name,
  product_name,
  resource_name,
  value
from
  scaleway_billing_consumption;

+----------------+---------------------------------+-----------------------------+-------------------------------------------------------+
| category_name  | product_name                    | resource_name               | value                                                 |
+----------------+---------------------------------+-----------------------------+-------------------------------------------------------+
| Compute        | <null>                          | Dynamic IP                  | {"currency_code":"EUR","nanos":480000000,"units":6}   |
| Object Storage | Multi-AZ Standard               | Multi-AZ - PAR              | {"currency_code":"EUR","nanos":270000000,"units":0}   |
| Containers     | Registry - Storage              | FR Storage                  | {"currency_code":"EUR","nanos":20000000,"units":0}    |
| Containers     | Kubernetes Kapsule              | Run - WAW                   | {"currency_code":"EUR","nanos":0,"units":0}           |
| Object Storage | Outgoing bandwidth free tier    | Pack                        | {"currency_code":"EUR","nanos":0,"units":0}           |
| Compute        | PRO2-S                          | PRO2-S - PL-WAW-1           | {"currency_code":"EUR","nanos":780000000,"units":354} |
| Serverless     | Serverless Containers - CPU     | CPU consumption - FR-PAR    | {"currency_code":"EUR","nanos":870000000,"units":8}   |
| Serverless     | Serverless Containers - Memory  | Memory consumption - FR-PAR | {"currency_code":"EUR","nanos":550000000,"units":1}   |
| Containers     | Registry - Storage              | PL Storage                  | {"currency_code":"EUR","nanos":20000000,"units":0}    |
| Observability  | Scaleway's logs                 | Scaleway's logs - FR-PAR    | {"currency_code":"EUR","nanos":0,"units":0}           |
| Serverless     | Serverless Containers Free Tier | Pack                        | {"currency_code":"EUR","nanos":0,"units":0}           |
| Object Storage | One Zone IA                     | ONEZONE IA - PAR            | {"currency_code":"EUR","nanos":20000000,"units":0}    |
| Object Storage | Internal Bandwidth              | Intra region fr-par         | {"currency_code":"EUR","nanos":0,"units":0}           |
| Observability  | Scaleway's samples              | Scaleway's samples - FR-PAR | {"currency_code":"EUR","nanos":0,"units":0}           |
| Storage        | Snapshot                        | Local storage               | {"currency_code":"EUR","nanos":380000000,"units":2}   |
| Compute        | Block Storage                   | Block Storage - WAW1        | {"currency_code":"EUR","nanos":960000000,"units":7}   |
| Observability  | Free plan                       | Free plan - FR-PAR          | {"currency_code":"EUR","nanos":0,"units":0}           |
+----------------+---------------------------------+-----------------------------+-------------------------------------------------------+

### CODE REVIEW

File: docs/tables/scaleway_billing_consumption.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+title: "Steampipe Table: scaleway_billing_consumtion - Query Scaleway Billing comsumption using SQL"
+description: "Allows users to query Scaleway Billing Consumption, specifically providing information about the billing"
+---
+
+# Table: scaleway_billing_consumtion - Query Scaleway Billing comsumption using SQL
+
+A Scaleway Billing Consumption track the costs of various scaleway products such as object_storage, storage, serverless, ect.
+
+This table requires an Organization ID to be configured in the scaleway.spc file.
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:10]:
```suggestion
**Important Notes**
- This table requires an Organization ID to be configured in the `scaleway.spc` file.
```

File: docs/tables/scaleway_billing_consumption.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+## Table Usage Guide
+
+The `scaleway_billing_consumtion` table offers insights into the Billing Consumption within Scaleway.
+
+## Examples
+
+### Basic info
+Explore the billing for all scaleway service.
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:19]:
```suggestion
Explore the billing for all scaleway services.
```

File: docs/tables/scaleway_billing_consumption.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+Explore costs by product category of scaleway
+
+```sql+postgres
+select 
+  category_name, 
+  sum((value -> 'units')::float) AS units 
+from 
+  scaleway_billing_consumption 
+group by 
+  category_name
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:53]:
```suggestion
  category_name;
```

File: docs/tables/scaleway_billing_consumption.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+```sql+sqlite
+select 
+  category_name, 
+  sum((value -> 'units')::float) AS units 
+from 
+  scaleway_billing_consumption 
+group by 
+  category_name
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:63]:
```suggestion
  category_name;
```

File: docs/tables/scaleway_billing_consumption.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+### List consumption by project_id
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:66]:
```suggestion
### List consumption by Project ID
```

File: docs/tables/scaleway_billing_consumption.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+Explore costs by product category of scaleway
+
+```sql+postgres
+select 
+  project_id, 
+  sum((value -> 'units')::float) AS units 
+from 
+  scaleway_billing_consumption 
+group by 
+  project_id
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:76]:
```suggestion
  project_id;
```

File: docs/tables/scaleway_billing_consumption.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+```sql+sqlite
+select 
+  project_id, 
+  sum((value -> 'units')::float) AS units
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:82]:
```suggestion
  sum((value -> 'units')::float) as units 
```

File: docs/tables/scaleway_billing_consumption.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+```sql+sqlite
+select 
+  project_id, 
+  sum((value -> 'units')::float) AS units 
+from 
+  scaleway_billing_consumption 
+group by 
+  project_id
```
Comment (misraved) [inline docs/tables/scaleway_billing_consumption.md:86]:
```suggestion
  project_id;
```

File: scaleway/table_scaleway_billing_consumption.go [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableScalewayBillingConsumption(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "scaleway_billing_consumption",
+		Description: "A Scaleway Billing Consumption.",
```
Comment (misraved) [inline scaleway/table_scaleway_billing_consumption.go:20]:
```suggestion
		Description: "Scaleway Billing Consumption",
```

File: scaleway/table_scaleway_billing_consumption.go [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+		List: &plugin.ListConfig{
+			Hydrate: listBillingConsumption,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "product_name",
+					Require: plugin.Optional,
```
Comment (misraved) [inline scaleway/table_scaleway_billing_consumption.go:26]: The code to handle the optional qual seems to be missing.

File: scaleway/table_scaleway_billing_consumption.go [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+	}
+}
+
+// // LIST FUNCTION
+func listBillingConsumption(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_project.listBillingConsumption", "connection_error", err)
```
Comment (misraved) [inline scaleway/table_scaleway_billing_consumption.go:82]:
```suggestion
		plugin.Logger(ctx).Error("scaleway_billing_consumption.listBillingConsumption", "connection_error", err)
```

File: scaleway/table_scaleway_billing_consumption.go [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	// Create SDK objects for Scaleway Billing
+	billingApi := billing.NewAPI(client)
+
+	// Get organisationID from config to request IAM API
+	organisationId := GetConfig(d.Connection).OrganizationID
+	if organisationId == nil {
+		err := fmt.Errorf("missing organization_id in scaleway.spc")
+		plugin.Logger(ctx).Error("scaleway_project.listBillingConsumption", "query_error", err)
```
Comment (misraved) [inline scaleway/table_scaleway_billing_consumption.go:93]:
```suggestion
		plugin.Logger(ctx).Error("scaleway_billing_consumption.listBillingConsumption", "query_error", err)
```

File: scaleway/table_scaleway_billing_consumption.go [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+		}
+	}
+	req.PageSize = scw.Uint32Ptr(uint32(maxResult))
+
+	var count int
+
+	for {
+		resp, err := billingApi.ListConsumptions(req)
+		if err != nil {
+			plugin.Logger(ctx).Error("scaleway_project.listBillingConsumption", "query_error", err)
```
Comment (misraved) [inline scaleway/table_scaleway_billing_consumption.go:119]:
```suggestion
			plugin.Logger(ctx).Error("scaleway_billing_consumption.listBillingConsumption", "query_error", err)
```

**Review by misraved — CHANGES_REQUESTED** [review]
@jplanckeel please take a look at the review comments. Thanks!!

File: scaleway/table_scaleway_billing_consumption.go (hunk 10 above)
Comment (jplanckeel) [reply scaleway/table_scaleway_billing_consumption.go:26]: i had this :)

**Review by misraved — APPROVED** [review]

**Comment by misraved** [comment]
Thanks @jplanckeel for making the changes 🎉 !!

**Pull request merged**